### PR TITLE
Drop support for Python 3.7

### LINKED ISSUE #355: Drop support for Python 3.7
### Problem Description

Python 3.7 reached its EOL June 27, 2023. As Github drops support some of our builds have started to fail. We should just drop support for 3.7 as a result.

### Expected behavior

- Drop support for 3.7 in setup.py
- Remove all 3.7 builds

### PR DESCRIPTION
**Pull request by amontanez24**
resolves #355

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -132,7 +131,7 @@
     keywords='copulas',
     name='copulas',
     packages=find_packages(include=['copulas', 'copulas.*']),
-    python_requires='>=3.7,<3.12',
+    python_requires='>=3.8,<3.12',
```
Comment (R-Palazzo) [inline setup.py:134]: Should we also update this line:
https://github.com/sdv-dev/Copulas/blob/a863cdd578d6d07d667a4a1d1bf44f69f5aa9ec8/setup.py#L15

**Review by R-Palazzo — COMMENTED** [review]
Looks good!
I found references of python 3.7 here:
https://github.com/sdv-dev/Copulas/blob/a863cdd578d6d07d667a4a1d1bf44f69f5aa9ec8/copulas/multivariate/vine.py#L75
Should we update the message?

**Review by R-Palazzo — APPROVED** [review]
LGTM!

**Review by pvk-developer — APPROVED** [review]
Missing `tox.ini` to be changed:
https://github.com/sdv-dev/Copulas/blob/a863cdd578d6d07d667a4a1d1bf44f69f5aa9ec8/tox.ini#L2

After that should be good :shipit:

**Pull request merged**